Dockerfile

```diff
@@ -49,15 +49,25 @@ COPY ./solidity $APP_DIR/solidity
 RUN cd $APP_DIR/solidity && npm install
 
 # Generate code.
-COPY ./pkg/chain/gen $APP_DIR/pkg/chain/gen
+COPY ./pkg/chain/gen/ethereum $APP_DIR/pkg/chain/gen/ethereum
 COPY ./pkg/ecdsa/tss/gen $APP_DIR/pkg/ecdsa/tss/gen
 # Need this to resolve imports in generated Ethereum commands.
 COPY ./config $APP_DIR/config
-RUN go generate ./.../gen
+RUN go generate ./...
 
 # Build the application.
 COPY ./ $APP_DIR/
 
+# Cleanup the `pkg/chain/gen` dir from unused chains bindings. Leave only
+# the ones which are currently in use. This helps reducing the size of
+# resulting binary and can prevent unexpected errors which may occur due to
+# transitive dependencies conflicts.
```

Comment:
Does it mean that both celo and ethereum contract bindings are generated? I do not follow why we can have problems with transitive dependencies.

Reply:
Both `pkg/chain/gen/ethereum` and `pkg/chain/gen/celo` are generated during `go generate ./...`. During `go build` only one directory is included in the resulting binary due to build tags. If we use, for example, the ethereum connector, celo bindings are not imported at all so there are not included.

However, there is a problem during testing. According to the [docs](https://golang.org/pkg/cmd/go/internal/test/), `go test` builds a testing binary for each package. That means for `pkg/chain` tests both `gen/ethereum` and `gen/celo` are included in the test binary. In that case they not conflict but themselves but `gen/celo` depends on `celo-blockchain` parts which import `celo-bls-go`. As I observed that binary has problems on Alpine and cause errors while [running tests](https://app.circleci.com/pipelines/github/keep-network/keep-ecdsa/3462/workflows/0c91fe7c-4028-48e3-a4fa-3701ea5b0d92/jobs/13585). This is why I added this line in the Dockerfile. It acts as a workaround. I plan to revisit this in the near future and create a proper Dockerfile for Celo or refactor this one to support multiple chains if possible.